### PR TITLE
Add feature flags and DIM_FLAVOR replacement

### PR DESCRIPTION
**Pull request by bhollis**
This PR suggests a pattern we could use to limit features based on some criteria. To start, I've set things up so we have a `$DIM_FLAVOR` replacement that is either not replaced (dev), "beta", or "release. Then there's a `dimFeatureFlags` object that I used to set tags+notes to be disabled in release only. I also hid the version info popup in dev.

### CODE REVIEW

File: app/scripts/dimApp.config.js [hunk 1]
```diff
@@ -54,10 +57,13 @@
         }
 
         console.log('DIM v$DIM_VERSION - Please report any errors to https://www.reddit.com/r/destinyitemmanager');
-        dimInfoService.show('changelogv$DIM_VERSION'.replace(/\./gi, ''), {
-          title: 'DIM v$DIM_VERSION Released',
-          view: 'views/changelog-toaster.html?v=v$DIM_VERSION'
-        });
+        // eslint-disable-next-line no-constant-condition
+        if ("$DIM_FLAVOR" === 'release' && "$DIM_FLAVOR" === 'beta') {
```
Comment (delphiactual) [inline app/scripts/dimApp.config.js]: shouldn't this be || not &&

**Comment by bhollis** [comment]
Totally

**Comment by kyleshay** [comment]
I like this solution @bhollis, thanks!!

**Comment by bhollis** [comment]
OK, I've put vendors and quality stats under the feature flag too. With this, we should be able to release from `dev` with no cherry-picking or anything.

As part of putting the quality stuff under the flag, I also fixed it so we won't show quality ratings in the loadout builder (we missed that before) and we only offer the "current" loadout calculation mode, not the "scaled to 335" mode. I also hid the "show advanced stat stuff" settings checkbox, so people aren't as confused that it's gone.

**Pull request merged**